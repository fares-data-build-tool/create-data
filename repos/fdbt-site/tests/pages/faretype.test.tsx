import * as React from 'react';
import { shallow } from 'enzyme';
import FareType from '../../src/pages/fareType';

describe('pages', () => {
    const mockErrors = [{ errorMessage: 'Choose a fare type from the options', id: 'fare-type-error' }];

    describe('fareType', () => {
        it('should render correctly', () => {
            const tree = shallow(<FareType errors={[]} />);
            expect(tree).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed to the page', () => {
            const tree = shallow(<FareType errors={mockErrors} />);
            expect(tree).toMatchSnapshot();
        });
    });
});
