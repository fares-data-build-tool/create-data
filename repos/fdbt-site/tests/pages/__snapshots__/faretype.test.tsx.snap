// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages fareType should render correctly 1`] = `
<Layout
  description="Fare Type selection page of the Fares data build tool"
  title="Fare Type - Fares data build tool"
>
  <main
    className="govuk-main-wrapper app-main-class"
    id="main-content"
    role="main"
  >
    <form
      action="/api/fareType"
      method="post"
    >
      <ErrorSummary
        errors={Array []}
      />
      <div
        className="govuk-form-group "
      >
        <fieldset
          aria-describedby="fareType-page-heading"
          className="govuk-fieldset"
        >
          <legend
            className="govuk-fieldset__legend govuk-fieldset__legend--xl"
          >
            <h1
              className="govuk-fieldset__heading"
              id="fareType-page-heading"
            >
              Select a fare type
            </h1>
          </legend>
          <FormElementWrapper
            errorClass="govuk-radios--error"
            errorId="fare-type-error"
            errors={Array []}
          >
            <div
              className="govuk-radios"
            >
              <div
                className="govuk-radios__item"
              >
                <input
                  className="govuk-radios__input"
                  id="fareType-single"
                  name="fareType"
                  type="radio"
                  value="single"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="fareType-single"
                >
                  Single - Point to Point
                </label>
              </div>
              <div
                className="govuk-radios__item"
              >
                <input
                  className="govuk-radios__input"
                  id="fareType-period"
                  name="fareType"
                  type="radio"
                  value="period"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="fareType-period"
                >
                  Period Tickets (Day, Week, Month and Annual)
                </label>
              </div>
              <div
                className="govuk-radios__item"
              >
                <input
                  className="govuk-radios__input"
                  id="fareType-return"
                  name="fareType"
                  type="radio"
                  value="return"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="fareType-return"
                >
                  Return - Single Service
                </label>
              </div>
            </div>
          </FormElementWrapper>
        </fieldset>
      </div>
      <input
        className="govuk-button govuk-button--start"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </form>
  </main>
</Layout>
`;
